### DISCUSSION TITLE
Automate deployment of docs app

### DISCUSSION THREAD

**Discussion by tmair** [ReactiveX/rxjs#6986] [Ideas / Feature request]
I want to suggest the more automation around the deployment of the docs app. It seems that the deployment is a step that has to be performed by @benlesh or other members of the core team manually, so some degree of automation could improve the amount of manual work associated with the documentation. 

My idea was to use github actions to deploy a new version of the docs app when the following events occur:
1. A new release is cut. The docs app should be deployed for the new release on the `live` channel
2. A merge to master is performed: The docs app should be deployed on the `next` channel

I had a look at the  [actions-hosting-deploy](https://github.com/FirebaseExtended/action-hosting-deploy) action. It seems that this could be a useful action for this.

Later on, it might also be beneficial to use the preview feature of Firebase to be able to see a preview of the docs app for a single PR, but that is something that I would only address if the main docs app automation is in place.

I would like to work on this, but before I dive into it I wanted to hear some opinions from the core team if this is something worth looking into.

**Comment by benlesh** [comment]
1. I wouldn't want to deploy any docs until the version is published.
2. We've had attempts to deploy docs automatically when we publish the library, but those attempts have failed in the past.

Otherwise, it sounds fine, but I'm not sure we'll have time to address it.

**Reply by tmair** [reply]
Thanks for the answer. As this will require some resources from the core team let's not do it. I wanted to ask the core team before puting any time into it.

**Comment by benlesh** [comment]
CORE TEAM: This is probably blocked on getting a monorepo in place.
